### PR TITLE
[doc] Remove "tsh db login" from database guides

### PR DESCRIPTION
**Pull request by greedy52**
Related PR
- #8270 (allows `tsh db connect` without `tsh db login`)
- #14336  (allows `tsh proxy db` without `tsh db login`)

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]
Nice simplification

File: docs/pages/database-access/architecture.mdx [hunk 1]
```diff
@@ -55,11 +55,11 @@ connect to a database.
    a client certificate. See [Issuing User Certificates](../architecture/authentication.mdx)
    for more details on how it works.
 2. The user picks the database they want to connect to from the list of available
-   databases shown in `tsh db ls` command and retrieves a short-lived X.509
-   certificate for it with `tsh db login`.
-3. The user uses a standard database client (e.g. `psql`, `mysql` or one of the
-   [graphical clients](./guides/gui-clients.mdx)) to connect to the Proxy, authenticating
-   with the client certificate from step 2.
+   databases shown in `tsh db ls` command.
+3. The user connects to the database with `tsh db connect` command which first
```
Comment (ptgott) [inline docs/pages/database-access/architecture.mdx:59]:
```suggestion
3. The user connects to the database with the `tsh db connect` command, which first
```

File: docs/pages/database-access/architecture.mdx [hunk 2]
```diff
@@ -84,8 +84,9 @@ Let's take a detailed look at each authentication point.
 
 ### Database client to Proxy
 
-Database clients authenticate with the Proxy using X.509 client certificates
-obtained from the `tsh db login` command.
+When running the `tsh db connect` command, it first retrieves a short-lived
```
Comment (ptgott) [inline docs/pages/database-access/architecture.mdx:87]: What is "it" here?

File: docs/pages/database-access/architecture.mdx [hunk 3]
```diff
@@ -96,15 +97,9 @@ obtained from the `tsh db login` command.
...
-The login command updates database-specific local configuration files (e.g.
-PostgreSQL [connection service file](https://www.postgresql.org/docs/13/libpq-pgservice.html)
-or MySQL [option file](https://dev.mysql.com/doc/refman/8.0/en/option-files.html)
-to group connection information for a particular database, which CLI clients
-can refer to.
-
-For configuring graphical clients, there is a `tsh db config` command that
-prints detailed information about the connection such as host/port and location
-of the secrets. See [GUI Clients](./guides/gui-clients.mdx) for details.
+For configuring graphical clients, use the `tsh proxy db` command which prints
```
Comment (ptgott) [inline docs/pages/database-access/architecture.mdx:100]:
```suggestion
For configuring graphical clients, use the `tsh proxy db` command, which prints
```

File: docs/pages/database-access/architecture.mdx [hunk 4]
```diff
@@ -96,15 +97,9 @@ obtained from the `tsh db login` command.
...
+detailed information about the connection such as host/port and location of the
```
Comment (ptgott) [inline docs/pages/database-access/architecture.mdx:101]:
```suggestion
detailed information about the connection such as the host, port, and location of the
```

File: docs/pages/database-access/guides/gui-clients.mdx [hunk 5]
```diff
@@ -79,15 +79,19 @@ You won't need to configure any credentials when connecting to this tunnel.
...
 
-You can then connect to the address the proxy command returns, in our example it
+You can optionally specify the database name and the user to use by default
+when connecting to the database:
+
+```code
+$ tsh proxy db --db-user=my-database-user --db-name=my-schema --tunnel <database-name>
+```
+
+Now, you can then connect to the address the proxy command returns, in our example it
```
Comment (ptgott) [inline docs/pages/database-access/guides/gui-clients.mdx:94]:
```suggestion
Now, you can connect to the address the proxy command returns. In our example it
```

File: docs/pages/includes/database-access/sql-server-local-proxy.mdx [hunk 6]
```diff
@@ -1,24 +0,0 @@
-SQL Server clients connect to a Teleport cluster via the local proxy in
```
Comment (ptgott) [inline docs/pages/includes/database-access/sql-server-local-proxy.mdx:1]: Just double-checking why this partial is being removed

Comment (greedy52) [reply docs/pages/includes/database-access/sql-server-local-proxy.mdx:1]: I couldn't find any page sourcing it. Seems referenced removed in this change: https://github.com/gravitational/teleport/pull/11801/files

File: docs/pages/database-access/architecture.mdx [hunk 7]
```diff
@@ -55,11 +55,11 @@ connect to a database.
...
-   databases shown in `tsh db ls` command and retrieves a short-lived X.509
-   certificate for it with `tsh db login`.
-3. The user uses a standard database client (e.g. `psql`, `mysql` or one of the
-   [graphical clients](./guides/gui-clients.mdx)) to connect to the Proxy, authenticating
-   with the client certificate from step 2.
+   databases shown in `tsh db ls` command.
+3. The user connects to the database with the `tsh db connect` command, which
+   first retrieves a short-lived X.509 certificate and then launches the
+   standard database client (e.g. `psql`) with this client certificate to
+   authenticate with the Proxy.
```
Comment (ptgott) [inline docs/pages/database-access/architecture.mdx:62]:
```suggestion
   authenticate with the Teleport Proxy Service.
```
To be a bit more precise, which architecture is often a place where new users get confused

**Review by ptgott — APPROVED** [review]
Approved with a small suggestion. You'll also need to fix the conflict (it's one line in the Snowflake guide).

**Pull request merged**